test/src/tests/search.test.js

```diff
@@ -0,0 +1,51 @@
+import {
+  CLIENT_URL,
+  uniqueId,
+  registerNewUser,
+  registerDandiset,
+} from '../util';
+import * as homePage from '../pages/homePage';
+import * as searchPage from '../pages/searchPage';
+
+
+describe('dandiset search page', () => {
+  const name = `test${uniqueId()}`;
+  const description = `Description ${uniqueId()}`;
+
+  beforeAll(async () => {
+    await page.goto(CLIENT_URL);
+    await registerNewUser();
+
+    // create a new dandiset that will appear in search results
+    await registerDandiset(name, description);
+  });
+
+  it('finds a dandiset that exists by name', async () => {
+    // search from the home page
+    await page.goto(CLIENT_URL);
+    await homePage.search(name);
+
+    // there should only be one result, the dandiset we just created
+    const results = await searchPage.getSearchResults();
+    expect(results).toHaveLength(1);
+    expect(results[0].name).toStrictEqual(name);
+  });
+
+  it('does not find a dandiset that does not exist', async () => {
+    // search for a nonexistent name
```

Comment:
This case (and several others) does not start with a page navigation, making it dependent on the success of the previous test. I think these sort of inter-test dependencies should be avoided if possible, so test failures are easier to diagnose.

Reply:
I think if there is a place to fudge this rule, it would be browser tests. Test setup is generally done through the UI, which is good, but also can be very slow. Having to to do lengthy setup steps for lots of really granular assertions can severely impact run time.
In this case, I think you're right, I need to refactor these to avoid intertest dependencies. Jest almost seems to encourage that pattern, but I should have known better.